var/spack/repos/builtin/packages/hdf5/package.py

```diff
@@ -70,23 +70,82 @@ class Hdf5(AutotoolsPackage):
     depends_on('szip', when='+szip')
     depends_on('zlib@1.1.2:')
 
-    @run_before('configure')
-    def validate(self):
-        """
-        Checks if incompatible variants have been activated at the same time
+    # According to ./configure --help thread-safe capabilities are:
+    # "Not compatible with the high-level library, Fortran, or C++ wrappers."
+    # (taken from hdf5@1.10.0patch1)
+    conflicts('+threadsafe', when='+cxx')
+    conflicts('+threadsafe', when='+fortran')
 
-        :param spec: spec of the package
-        :raises RuntimeError: in case of inconsistencies
+    @property
+    def libs(self):
+        """Hdf5 can be queried for the following parameters:
+        
+        - "hl": high-level interface
+        - "cxx": C++ APIs
+        - "fortran": fortran APIs
+        
+        :return: list of matching libraries
         """
+        query_parameters = self.spec.last_query.extra_parameters
+
+        shared = bool('+shared' in self.spec)
+
+        # This map contains a translation from query_parameters
+        # to the libraries needed
+        query2libraries = {
+            tuple(): ['libhdf5'],
+            ('cxx', 'fortran', 'hl'): [
```

Comment:
so you need to do all the possible combinations here between `cxx`, `fortran` and `hl`? Does Python throw an error in `libraries = query2libraries[key]` if you happen to forget to cover some combinations or if package developer made a typo and queries for `fortran`?

Also does order matter, i.e. `cxx`,`hl` vs `hl`,`cxx`.

p.s. I am a python noob, i am looking at this from C++ perspective :smile: